zerver/lib/request.py

```diff
@@ -367,6 +369,18 @@ def _wrapped_view_func(request: HttpRequest, *args: Any, **kwargs: Any) -> HttpR
 
             kwargs[func_var_name] = val
 
+        request._unsupported_params = []
+
+        if hasattr(request, 'method'):
+            req_params = set()
+            if request.method == 'POST':
+                req_params = set(request.POST.keys())
+            elif request.method == 'GET':
+                req_params = set(request.GET.keys())
```

Comment:
Given that we allow both POST and GET parameter formats in many cases, I'd prefer to just union the sets before doing the subtraction.  I think that's going to produce better clarity for the user in any case -- if we ignore their GET parameter because we only allow POST parameters, we should report it.

Otherwise the `request.py` logic looks good to me.